### ISSUE TITLE
Check why AdGuard is not listed in Google's apps backup

### ISSUE THREAD

**Issue by ameshkov** [AdguardTeam/AdguardForAndroid#1396]
Talking about this:
https://developer.android.com/guide/topics/data/autobackup.html#Files

In theory, it should be available for autobackup.

**Comment by nkartyshov** [comment]
By default android do not backuping a app working in foreground.

>During Auto Backup, the system shuts down the app to make sure it is no longer writing to the file system. By default, the backup system ignores apps that are running in the foreground because users would notice their apps being shut down. 

But we can enable backup using the parameter android:backupInForeground

**Comment by ameshkov** [comment]
The question is why don't I see AG in the list of apps available for auto-backup.

**Comment by nkartyshov** [comment]
Everything very simple, Adguard backup have more 25Mb size and android cannot backuping Adguard. 
We can exclude files that should not be in a backup e.g. filters.

**Comment by TPS** [comment]
&/or give the Auto Backup a _compressed_ version of all this text to keep ≤ 25 MB?

**Comment by ameshkov** [comment]
@nkartyshov if we exclude files what will be the backup size then?

@TPS files aren't required, really. This is more like an app cache, who needs this.

**Comment by nkartyshov** [comment]
Excluding a folder file solved the problem.

**Comment by carlylemiii** [comment]
Was this implemented? I don't see it in my list of backed up apps, either (neither viewing my backup in Google Drive nor backed up apps in Android Settings).

**Comment by ameshkov** [comment]
@carlylemiii it was, but I do not see it as well, reopening the issue.

**Comment by ameshkov** [comment]
Oh, so auto-backup does not exclude external cache dir by default?

I guess we should exclude the whole `getExternalFilesDir()`

**Comment by nkartyshov** [comment]
@ameshkov Yes, auto-backup includes a internal files directory and an external files directory by default.

**Comment by nkartyshov** [comment]
Excluded the external folder with the auto-backup.

**Comment by carlylemiii** [comment]
Still not seeing AdGuard backed up.

**Comment by ameshkov** [comment]
Reopened, reassigned to v2.11

**Comment by nkartyshov** [comment]
The new implementation works fine, but I want to wait for a feedback after the release.

**Comment by carlylemiii** [comment]
:-)

<img width="274" alt="2017-10-15 15_49_53-photo - google photos" src="https://user-images.githubusercontent.com/29067983/31589079-b5c4bd80-b1c0-11e7-8c0e-451aa239f368.png">

**Comment by ameshkov** [comment]
Finally:)

Mind trying to uninstall/restore it?

**Comment by carlylemiii** [comment]
Worked with the 2.10.136 apk from earlier today! The only thing is the tutorial still launches, and then after skipping it, the main screen displays "get premium" for about ten seconds or so before it automatically figures out you had a license previously. I'm not sure if those are both expected, but I figured I'd point it out. But enabled filters, HTTPS, user filter, etc. all came back! Yay!

**Comment by ameshkov** [comment]
👍  thanks for testing it!

Regarding the license stuff, it is expected. For this backup AG now uses the very same protocol as for "export/import settings" options, and we don't export license key (just in case if people share their settings with anybody else). 

Actually, maybe we'd let private stuff backup as well. @nkartyshov please take a look.

**Comment by TPS** [comment]
@carlylemiii What precisely are you using to check app backup?

I can confirm I do ___NOT___ see (Amazon) AG in my GDrive app backup.

**Comment by carlylemiii** [comment]
System > Backup > App data, on my Google Pixel, with Adguard 2.10.136, but the timestamp from my backup I was probably on 2.10.133.

**Comment by TPS** [comment]
Well, I've only .108 & nothing yet on Amazon past that, so that's probably it.

@carlylemiii I haven't that menu option precisely, but the GDrive backup points back to my closest version, so I think we're watching the same thing.

**Comment by carlylemiii** [comment]
It does show in my Google Drive as well (unsurprisingly). Note that there it's actually after most other apps (comes after YouTube, in my case). It shows there as com.adguard.android, at the very beginning of a long list of com.__ packages, and does not have the AdGuard icon, just a generic cloud backup icon.

**Comment by TPS** [comment]
My last item is (a quite generic) F-droid listing, so I'm reasonably sure it's just my lagging version.

**Comment by ameshkov** [comment]
Starting from the next update it'll include license stuff

**Comment by zero77** [comment]
@ameshkov 
This is no longer working as the AdGuard app is not listed in backups.

This may be of help:
https://developer.android.com/guide/topics/data/autobackup
https://developer.android.com/identity/data/autobackup

**Comment by Versty** [comment]
@zero77 Hi! 

Backups have been disabled starting from v4.0.
Could you please create a separate issue (Feature Request) if you’d like this option to be brought back?

**Comment by TPS** [comment]
@Versty I'm sorry: why, on both counts?
- Why was a working feature removed w/ 0 notice?
- If there's an existing issue (with history!) for it that already has supporters tracking it, why not simply reopen that than duplicating unnecessarily?

**Comment by Versty** [comment]
@TPS 

This issue is related to earlier versions of the application.
In v4.0, we completely reworked AdGuard, so it would be more convenient to discuss this separately. The history can be added as a link to this issue.

Backups were replaced with extended export/import functionality, but we’re currently discussing the possibility of bringing them back if this feature is widely requested by users.

**Comment by zero77** [comment]
As requested I have opened another issue.
https://github.com/AdguardTeam/AdguardForAndroid/issues/5879

**Comment by TPS** [comment]
Reimplemented in #5879
